jax/experimental/maps.py

```diff
@@ -1081,8 +1081,50 @@ def _dynamic_jaxpr_process_xmap(self, primitive, f, tracers, params):
   self.frame.eqns.append(eqn)
   return out_tracers
 pe.DynamicJaxprTrace.process_xmap = _dynamic_jaxpr_process_xmap  # type: ignore
+
+def _xmap_partial_eval_custom_params_updater(
+    unks_in: Sequence[bool],
+    kept_outs_known: Sequence[bool], kept_outs_staged: Sequence[bool],
+    num_res: int, params_known: dict, params_staged: dict
+  ) -> Tuple[dict, dict]:
+  assert params_known['spmd_in_axes'] is None and params_known['spmd_out_axes'] is None
+  assert params_staged['spmd_in_axes'] is None and params_staged['spmd_out_axes'] is None
+
+  # pruned inputs to jaxpr_known according to unks_in
+  donated_invars_known, _ = pe.partition_list(unks_in, params_known['donated_invars'])
+  in_axes_known, _ = pe.partition_list(unks_in, params_known['in_axes'])
+  residual_axes: Sequence[AxisNamePos]
+  if num_res == 0:
+    residual_axes = ()
```

Comment:
```suggestion
    residual_axes = []
```

If you make this an empty list, then you can delete the type annotation a couple lines above, right?